### ISSUE TITLE
Bogus indentation warning for nested SELECT

### ISSUE THREAD

**Issue by barrywhart** [alanmcruickshank/sqlfluff#204]
Input:
```
SELECT
    user_id
FROM (
    SELECT
        c.user_id AS user_id
    FROM
        `c`
)
```

Result:
```
$ sqlfluff lint --exclude-rules L001 --dialect bigquery test.sql
== [test.sql] FAIL
L:   8 | P:   1 | L003 | Indent expected and not found compared to line #3
```

I think the closing parenthesis is indented correctly, since it closes the indentation begun on the line `FROM (`. What's the intended indentation style for this situation?

**Comment by barrywhart** [comment]
Similar -- this query generates no warning, but only after indenting the `)` for the nested `SELECT` to align with the `SELECT`, rather than the opening `(`. I also tried placing the `(` and `)` in column 1, but that also generates a warning.

```
SELECT
    user_id
FROM
    `t1`
FULL JOIN (
    SELECT
        user_id,
        TRUE AS exists_in_user_map
    FROM
        `t2`
    )
    USING (user_id)
```

**Comment by alanmcruickshank** [comment]
Agree. I'm not sure how this one is happening though, will need to dive into things a bit more.

**Comment by alanmcruickshank** [comment]
Confirmed. This is happening because of how we deal with ambiguous indentation.

Technically the following is the *most indented* version which would be acceptable:

```
SELECT
    user_id
FROM
    (
        SELECT
            c.user_id AS user_id
        FROM
            c
    )
```

Note that there's an indent after `FROM` *and* after `(`. It's because of the two possible places for indentation that it's getting confused with the last bracket placement. I think however that the two queries you posted *are correct*, and that because the indent immediately after `FROM` wasn't used, then we shouldn't expect it's corresponding dedent to be present either.

I'll pick this bug up next.

**Comment by barrywhart** [comment]
Great analysis, thanks! 🙏

**Comment by alanmcruickshank** [comment]
Haha! I did a load of work on this one, thinking it was a massive issue. I think it's actually due to _closing brackets_ being a special case for indentation which I wasn't handling. PR coming up now.
